### DISCUSSION TITLE
Is it possible to use ASQL in a Authentication Strore implementation ?

### DISCUSSION THREAD

**Discussion by vingeni** [cutelyst/cutelyst#341] [Q&A]
Hi !
I was wondering how is it possible to use ASql with the Cutelyst Authentication Store ?
I haven't found any example.
Is it a possible use-case ?

**Comment by dantti** [comment]
It is and I use, it was not meant to be used in async mode so what I do is add a AuthenticationUser member for the Store class, and set it once I fetch the data with ASql, and then I call Authenticate with the Store find user returning that user, just make sure you always set a  user and don't do any fancy thread sharing of the Store object.

**Reply by vingeni** [reply]
Hi,

Do you make the ASql request into the Login::index controler ([like in the authentication tutorial](https://github.com/cutelyst/cutelyst/wiki/Tutorial_05_Authentication)) ?

Does it make sense to use Cutelyst::StoreMinimal for that ? or it is better to make my own, even more minimal, store class ?

would you be interested by a sample code for completing the tutorial ?

**Reply by dantti** [reply]
> Hi,
> 
> Do you make the ASql request into the Login::index controler ([like in the authentication tutorial](https://github.com/cutelyst/cutelyst/wiki/Tutorial_05_Authentication)) ?

Yes

> Does it make sense to use Cutelyst::StoreMinimal for that ? or it is better to make my own, even more minimal, store class ?

I find better make your own

> would you be interested by a sample code for completing the tutorial ?

sure, I need to review & improve it, but time is short as always

**Reply by vingeni** [reply]
I made this sample app on my space : 
https://github.com/vingeni/cutelyst-async-login
